tests/unit/test_repository.py

```diff
@@ -0,0 +1,467 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import os
+import random
+import string
+from contextlib import ExitStack as does_not_raise
+from typing import Optional
+
+import pytest
+
+import zenml
+from zenml.artifact_stores import LocalArtifactStore
+from zenml.enums import StackComponentType
+from zenml.exceptions import (
+    InitializationException,
+    RepositoryNotFoundError,
+    StackComponentExistsError,
+    StackExistsError,
+)
+from zenml.io import fileio
+from zenml.metadata_stores import MySQLMetadataStore, SQLiteMetadataStore
+from zenml.orchestrators import LocalOrchestrator
+from zenml.repository import Repository
+from zenml.stack import Stack
+
+
+def _create_local_stack(
+    stack_name: str,
+    orchestrator_name: Optional[str] = None,
+    metadata_store_name: Optional[str] = None,
+    artifact_store_name: Optional[str] = None,
+):
+    """Creates a local stack with components with the given names. If the
+    names are not given, a random string is used instead."""
+
+    def _random_name():
```

Comment:
How would you use hypothesis in this case when it is not an actual test function but a helper function?

Reply:
You'd get rid of the helper function and instead use the hypothesis decorator on the test and pass whatever random string(s) were generated by hypothesis. You'd need to make sure to pass in a `min_size=1` along with the [`text()` strategy](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.text).

Reply:
I tried but it doesn't work in combination with our clean repo fixture

Reply:
```hypothesis.errors.FailedHealthCheck: Function-scoped fixture 'clean_repo' used by 'tests/unit/test_repository.py::test_registering_a_stack'
E                               
E                               Function-scoped fixtures are not reset between examples generated by
E                               `@given(...)`, which is often surprising and can cause subtle test bugs.
E                               
E                               If you were expecting the fixture to run separately for each generated example,
E                               then unfortunately you will need to find a different way to achieve your goal
E                               (e.g. using a similar context manager instead of a fixture).```

Reply:
Ah yes, I'd forgotten about that. Do you hit a test strategy scope error?

Reply:
The error was called FailedHealthCheck

Reply:
Not sure if this applies in this case, but you can change the scope with `@pytest.fixture(scope="module")`

Reply:
Unfortunately not, this fixture creates a clean repo for one specific test so has to stay with function scope. Anyway I think the random strings in this case don't really matter as it's just for the names, but I agree we should definitely use this in cases where the strings are more relevant!